common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/FetchShuffleBlockChunks.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.shuffle.protocol;
+
+import java.util.Arrays;
+
+import io.netty.buffer.ByteBuf;
+
+import org.apache.spark.network.protocol.Encoders;
+
+// Needed by ScalaDoc. See SPARK-7726
+import static org.apache.spark.network.shuffle.protocol.BlockTransferMessage.Type;
+
+
+/**
+ * Request to read a set of block chunks. Returns {@link StreamHandle}.
+ *
+ * @since 3.2.0
+ */
+public class FetchShuffleBlockChunks extends AbstractFetchShuffleBlocks {
+  // The length of reduceIds must equal to chunkIds.size().
```

Comment:
How strong is this assumption ? Do we see a future evolution where this can break ? Or is it tied to the protocol in nontrivial ways ?
As an example - `encode` and `decode` do not assume this currently (we could have avoided writing `chunkIdsLen` if they did)

Reply:
This is a strong assumption. For a `reduceIds[i]`, each chunkId in chunksIds[i] represents a shuffleChunk  `shuffleChunk_<shuffleId>_reduceIds[i]_chunksIds[i][j]` which is being requested.
This is similar to the existing FetchShuffleBlock message. There are assertions on the server side as well to check this.

> Do we see a future evolution where this can break

Not really.  At least I can't think of any that would change this. This just represents all the chunkIds for a particular reduceIds that are being requested which I don't think will change.

> encode and decode do not assume this currently 

That's right. I can avoid writing `chunkIdsLen`. It is just that I mimicked the code in `FetchShuffleBlocks` which does the same.

Reply:
In general, given protocol evolution issues, I am fine with sending the length as part of the serde.
I was trying to understand if this assumption is very strict or whether things can evolve in future such that this assumption is broken.

Will resolve the conversation, thanks for clarifying.